src/main/include/subsystems/Drivetrain.h

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Team OKC Robotics
```

Comment:
alright tell me if I'm wrong
 * the header file here declares all the variables and classes and whatnot and the actual .cpp is just for method implementation?
 * so this is the actual Drivetrain subsystem class file.
 * the drivetrain interface is completely separate from the HardwareInterface thingy and so is useless? (or negates the need for the others?) or it's a subclass? implementation?
 *  the drivetrain subsystem class uses the drivetrain interface to access the hardware. not sure if we need another layer in there, but if it works it works.
 * all of the methods are stubs (or whatever they're called in header files) and are just copied/pasted names from the Java version of the code

Reply:
- The header file declares the class functions and variables that are implemented by a cpp file
- Yes
- The drivetrain interface serves as a way for us to access the relevant centralized hardware, which is declared and initialized in the RobotContainer. This is a struct so we don't need to pass a ton of variables in the constructor
- Yes that's exactly it. There doesn't need to be another layer though.
- yeah pretty much